src/ebfm/INIT.py

```diff
@@ -328,6 +328,79 @@ def init_grid(grid, io, config: GridConfig):
+        # -----------------------------------------------------------------------------------------------------
+        grid["nr_az_steps"] = 24  # number of azimuth angles (e.g. 24 = 1 per hour)
+
+        # azimuth angles in radians from -pi to +pi with nr_az_steps number of steps
+        grid["az_array"] = -np.pi * (-1.0 + 2.0 * (np.arange(1, grid["nr_az_steps"] + 1) / grid["nr_az_steps"]))
+
+        xl, yl = grid["x_2D"].shape
+
+        # loop over the azimuth angles to determine gridded maximum grid angles per angle
+        grid["maxgridangle"] = np.zeros((grid["gpsum"], grid["nr_az_steps"]), dtype=np.float64)
+        for n in range(grid["nr_az_steps"]):
+            az = np.full(int(grid["gpsum"]), grid["az_array"][n], dtype=float)
+
+            # calculate step sizes (ddx, ddy) in x- and y-directions for all azimuth angles
+            ddx = np.empty_like(az, dtype=float)
+            ddy = np.empty_like(az, dtype=float)
+            m = az <= -0.75 * np.pi
+            ddx[m] = -np.tan(np.pi + az[m])
+            m = (az <= -0.25 * np.pi) & (az > -0.75 * np.pi)
+            ddx[m] = -1.0
+            m = (az <= 0.25 * np.pi) & (az > -0.25 * np.pi)
+            ddx[m] = np.tan(az[m])
+            m = (az <= 0.75 * np.pi) & (az > 0.25 * np.pi)
+            ddx[m] = 1.0
+            m = az > 0.75 * np.pi
+            ddx[m] = np.tan(np.pi - az[m])
+            m = az <= -0.75 * np.pi
+            ddy[m] = 1.0
+            m = (az <= -0.25 * np.pi) & (az > -0.75 * np.pi)
+            ddy[m] = -np.tan(0.5 * np.pi + az[m])
+            m = (az <= 0.25 * np.pi) & (az > -0.25 * np.pi)
+            ddy[m] = -1.0
+            m = (az <= 0.75 * np.pi) & (az > 0.25 * np.pi)
+            ddy[m] = -np.tan(0.5 * np.pi - az[m])
+            m = az > 0.75 * np.pi
+            ddy[m] = 1.0
+
+            # from every grid cell step in the direction of the azimuth until the grid end is reached
+            # and detect maximum grid angle along the path
+            i0, j0 = np.where(mask_2D == 1)
+            max_angle = np.full(grid["gpsum"], -np.inf, dtype=np.float64)
+            count = 1
+            active = np.ones(grid["gpsum"], dtype=bool)
+            while active.any():
+                j = np.round(j0 + ddx * count).astype(np.int64)
+                i = np.round(i0 + ddy * count).astype(np.int64)
+
+                inbound = (j >= 0) & (j < yl) & (i >= 0) & (i < xl) & active
+                if not inbound.any():  # stop when all walks have reached the domain edge
+                    break
+
+                iv = i[inbound]
+                jv = j[inbound]
+
+                dx = grid["x_2D"][iv, jv] - grid["x"][inbound]
+                dy = grid["y_2D"][iv, jv] - grid["y"][inbound]
+                distance = np.hypot(dx, dy)  # walk distance from start to target
+
+                dz = grid["z_2D"][iv, jv] - grid["z"][inbound]
+                grid_angle = np.arctan(dz / distance)  # grid angle from start to target
```

Comment:
```suggestion
                grid_angle_proxy = np.arctan(dz**2 / distance_sq)  # grid angle from start to target
```
If we want to use squared distance we also have to use the squared `dz`.